src/mem/allocconfig.h

```diff
@@ -10,6 +10,17 @@ namespace snmalloc
 #  define CHECK_CLIENT
 #endif
 
+  SNMALLOC_FAST_PATH void check_client(bool test, const char* const str)
```

Comment:
This seems to rely on inlining and DCE to avoid evaluating the test expression in non-`CHECK_CLIENT` builds.  Does `SNMALLOC_FAST_PATH` ensure that all that takes place?  Maybe we should be shouting (that is, using a macro, perhaps `SNMALLOC_CHECK_CLIENT(,)` like `SNMALLOC_ASSERT`) to be sure the compiler understands? ;)

Either way is, to be clear, much nicer than all the clutter from the `#ifdef CHECK_CLIENT`s.